bdb/cursor.c

```diff
@@ -2021,6 +2023,7 @@ int truncate_asof_pglogs(bdb_state_type *bdb_state, int file, int offset)
         return_pglogs_commit_list(lcommit);
         lcommit = LISTC_BOT(&pglogs_commit_list);
     }
```

Comment:
This won't get called at all if new snapshot or pglogs isn't enabled .. I guess either move this line to above that check, or maybe add it to comdb2_recovery_cleanup directly ..

Reply:
I guess this same argument would go against piggy-backing recover_logfile_pglogs for PIT txns.   It's not part of this pr anyway but wanted to write the thought.  :)

Reply:
[Moved commit lsn truncation to its own function](https://github.com/bloomberg/comdb2/pull/3803/files#diff-91454b1e6e0271e4ab1900d4829d866b37f219f943ed9864fe0f2f23e4cc91e2R6041).